File: client/types/sys/Menu.ts
export type MenuItem = {
  id: number;
  name: string;
  engName?: string;
  url: string;
  icon?: string;
  displayOrder: number;
  isActive: boolean;
  parentId?: number;
  sysmodule?: string;
  children?: MenuItem[];
};

export const defaultMenuItem: MenuItem = {
  id: 0,
  name: '',
  engName: '',
  url: '',
  icon: '',
  displayOrder: 0,
  isActive: true,
  parentId: undefined,
  sysmodule: undefined,
  children: [],
};

export type SaveMenuDto = {
  id: number;
  name: string;
  engName?: string;
  url: string;
  icon?: string;
  displayOrder: number;
  isActive: boolean;
  parentId?: number;
  sysmodule: string;
};

export const defaultSaveMenuDto: SaveMenuDto = {
  id: 0,
  name: '',
  engName: '',
  url: '',
  icon: '',
  displayOrder: 0,
  isActive: true,
  parentId: undefined,
  sysmodule: '',
};

File: server/Model/Entities/System/Menu.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

// ...existing using statements...

namespace Model.Entities.System;

[Table("Menus")]
public class Menu : BaseEntity<int>
{
    [Required] [StringLength(100)] public string Name { get; set; } = string.Empty;
    [Required] [StringLength(100)] public string EngName { get; set; } = string.Empty;
    [Required] [StringLength(200)] public string Url { get; set; } = string.Empty;

    [Required] [StringLength(100)] public string? Icon { get; set; }

    public int DisplayOrder { get; set; }

    public bool IsActive { get; set; } = true;

    public int? ParentId { get; set; }

    [StringLength(100)] public string? Sysmodule { get; set; }
}